tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -1050,7 +1068,11 @@ static uint8_t bap_send(const void *cmd, uint16_t cmd_len,
 	(void)bt_conn_get_info(conn, &conn_info);
 
 	stream = stream_find(audio_conn, cp->ase_id);
-	if (stream == NULL || stream->stream.ep->dir != BT_AUDIO_DIR_SINK) {
+	if (stream == NULL ||
+	    (conn_info.role == BT_HCI_ROLE_CENTRAL &&
+	     stream->stream.ep->dir != BT_AUDIO_DIR_SINK) ||
+	    (conn_info.role == BT_HCI_ROLE_PERIPHERAL &&
+	     stream->stream.ep->dir != BT_AUDIO_DIR_SOURCE)) {
```

Comment:
The `bt_bap_ep_info` now contains a `can_send` field to make it easier to verify if you can TX on a given stream/ep